src/lib/rnp.cpp

```diff
@@ -2317,6 +2317,9 @@ try {
                                        get_key_prefer_public(handle),
                                        &handle->ffi->key_provider,
                                        PGP_KF_ENCRYPT);
+    if (!key) {
```

Comment:
Please add or update existing test which checks this case (i.e., for instance, attempt to add sign-only primary key with invalid or expired subkey for encryption), and check for failure of `rnp_op_encrypt_execute()` afterwards. That way we'll be sure that behaviour doesn't change in future/there are no memory leaks or other failures if incorrect key used for encryption attempt.

Reply:
@ni4 this case is already covered in https://github.com/rnpgp/rnp/blob/master/src/tests/ffi-enc.cpp#L852 - sign-only primary key and expired subkey

Reply:
@antonsviridenko Thanks, then no new test is needed.
P.S. Don't be shocked by all red CI - it's because of key expiration used in tests, marking key as expired. I'll issue PR fixing that.